### PR TITLE
WFLY-18128: correct declared licenses

### PR DESCRIPTION
**Pull request by michpetrov**
Issue: [WFLY-18128](https://issues.redhat.com/browse/WFLY-18128)

The "big" change here is replacing "the antlr x.y.z license" with the standard BSD-3-Clause, as that seems to be what [they're actually using](https://www.antlr.org/license.html). If anyone knows of a reason to include the special license I'll put it back.

Also, I've removed the `org.reactivestreams:reactive-streams` from the "galleon-pack/galleon-shared" package (the github diff obfuscates it a little), since it is already defined in "ee-feature-pack/galleon-shared". That's the only removal of an entire artifact record.

### CODE REVIEW

**Comment by yersan** [comment]
@michpetrov FYI, this got a conflict, could you rebase it?

**Review by yersan — CHANGES_REQUESTED** [review]
@michpetrov How we are verifying each of these changes? 

There are several artifacts with license changes, so I wondering how we are getting this license information

**Comment by michpetrov** [comment]
@yersan I've ran the build without using the license file config (that way the plugin is pulling the licenses from POMs) and compared the output to the normal build. I have a script for comparing the files if you want to see it.

**Comment by yersan** [comment]
@michpetrov Ok, I'm commenting on it because it seems is not always getting the expected result; I haven't checked every license but it seems that the one for Jakarta JPA is incorrect, could you take a look by doing another round? Thanks!

File: ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml [hunk 1]
```diff
@@ -931,8 +956,8 @@
                     <distribution>repo</distribution>
                 </license>
                 <license>
-                    <name>Eclipse Distribution License, Version 1.0</name>
-                    <url>http://repository.jboss.org/licenses/edl-1.0.txt</url>
+                    <name>GNU General Public License v2.0 only, with Classpath exception</name>
+                    <url>http://repository.jboss.org/licenses/gpl-2.0-ce.txt</url>
```
[the remark this answers is not in the document]
Comment (michpetrov) [reply ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml]: My bad, it's the resource-api, not persistence-api, that that's under EPL/GPL - https://projects.eclipse.org/projects/ee4j.jca

**Comment by michpetrov** [comment]
@yersan rebased

**Comment by michpetrov** [comment]
actually let me check it again, I think I skipped something

**Comment by michpetrov** [comment]
@yersan should be ok now

**Review by yersan — CHANGES_REQUESTED** [review]
@michpetrov These additional two look wrong, could you check them?

~* jakarta.resource:jakarta.resource~
~https://github.com/wildfly/wildfly/blob/81b92ecdc3a04e7844739147a34b5f590d67927d/ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml#L949-L964~
~See: https://projects.eclipse.org/projects/ee4j.jca~

My mistake, the above is indeed correct. So ignore it

* org.jboss.hal:hal-console https://github.com/wildfly/wildfly/blob/81b92ecdc3a04e7844739147a34b5f590d67927d/ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml#L3906-L3921
See: https://github.com/hal/console/blob/main/LICENSE

**Review by yersan — APPROVED** [review]
Giving my approval, with the pending detail about confirming if we should remove MIT license from HAL. Maybe @hpehl can confirm it?

The other points about antlr and reactive, look correct to me too, thanks @michpetrov!

**Comment by hpehl** [comment]
@yersan, yes, we can remove the MIT license from HAL.

**Comment by michpetrov** [comment]
@yersan rebased

**Comment by bstansberry** [comment]
Thanks @michpetrov and @yersan

**Pull request merged**